### DISCUSSION TITLE
iOS Build keeps getting rejected with ITMS-90424: Invalid Swift Support

### DISCUSSION THREAD

**Discussion by ajeetschouhan** [codemagic-ci-cd/codemagic-docs#1757] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

While using CodeMagic, whenever I am trying to deploy to app store, the build keeps getting rejected with ITMS-90424: Invalid Swift Support email.
If I try to run the build locally and upload to app store, it works without any issue.
Is there any config changes that I need to make in codemagic? 

### Expected results

Codemagic build should upload to app store without errors.

### Actual results

Codemagic build gets rejected with ITMS-90424: Invalid Swift Support

**Comment by himesh-cm** [comment]
Hi @ajeetschouhan, When you build locally and upload, do you get the rejection email?
What version of Xcode are you using? Can you share a build id as well?
Make sure to check this [SO](https://stackoverflow.com/questions/59728413/invalid-swift-support-the-swiftsupport-folder-is-empty) link.

**Reply by ajeetschouhan** [reply]
Hi @himesh-cm 

My last failed build id is: 6486d6e602db7cf5b46c809e

That build is having another issue with android as well where the android version is not picking up from pubspec.yaml so please ignore that I have another thread going on for that.

Back to iOS issue, building locally and uploading works for me. There is no rejection in that. Only when the builds are happening in codemagic, they are getting rejected with ITMS-90426: Invalid Swift Support - The SwiftSupport folder is missing. Rebuild your app using the current public (GM) version of Xcode and resubmit it.

Local version of XCode that works is: 14.3.1 (14E300c)
Same version is being used in codemagic as well (Latest(14.3.1))

**Reply by himesh-cm** [reply]
Hi @ajeetschouhan, Apologies for the delayed reply for the `swift support` Check [here](https://stackoverflow.com/a/42128128) if it could solve the issue.

**Reply by himesh-cm** [reply]
Also, the build id `6486d6e602db7cf5b46c809e` uses `ad-hoc` profile; you need `IOS_APP_STORE` when you want to publish the app store.
You can do it by going to `Workflow Editor Settings > Distribution > Provisioning Profile > App store`.
Please let me know if you run into any issues.

**Reply by himesh-cm** [reply]
Hi @ajeetschouhan, Did you get a chance to try it per my last message?

**Comment by himesh-cm** [comment]
Hi @ajeetschouhan, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
